crates/autopilot/src/solvable_orders.rs

```diff
@@ -391,6 +395,36 @@ fn orders_with_balance(
     orders
 }
 
+/// Filters out dust orders i.e. partially fillable orders that, when scaled
+/// have a 0 buy or sell amount.
+fn filter_dust_orders(mut orders: Vec<Order>, balances: &Balances) -> Vec<Order> {
+    orders.retain(|order| {
+        if !order.data.partially_fillable {
+            return true;
+        }
+
+        let Some(balance) = balances.get(&Query::from_order(order)) else {
+            return false;
+        };
```

Comment:
Doesn't this make the insufficient balance check that we have after the call to this function obsolete? And it doesn't have the special ethflow logic from that check.

Reply:
> Doesn't this make the insufficient balance check that we have after the call to this function obsolete?

No, this just checks that balance scaling doesn't yield a different amount, and only for partially fillable orders. I intentionally wanted to keep the filtering separate so that they are logged as being filtered for different reasons. I changed the filter to be applied _after_ the balance filtering to be more accurate.

> And it doesn't have the special ethflow logic from that check.

Good catch. Fixed.